lxd/storage/storage.go

```diff
@@ -4,9 +4,93 @@ import (
 	"os"
 
 	"github.com/lxc/lxd/lxd/project"
+	"github.com/lxc/lxd/lxd/state"
 	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
 )
 
+var backends = map[string]driver{
+	"dir": &dir{},
+}
+
+// VolumeType defines the type of a volume
+type VolumeType int
+
+const (
+	// VolumeTypeContainer represents the container type.
+	VolumeTypeContainer VolumeType = iota
+	// VolumeTypeContainerSnapshot represents the container snapshot type.
+	VolumeTypeContainerSnapshot
+	// VolumeTypeCustom represents a custom volume type.
+	VolumeTypeCustom
+	// VolumeTypeCustomSnapshot represents a custom volume snapshot type.
+	VolumeTypeCustomSnapshot
+	// VolumeTypeImage represents an image type.
+	VolumeTypeImage
+	// VolumeTypeImageSnapshot represents an image snapshot type.
+	VolumeTypeImageSnapshot
+)
+
+type driver interface {
+	Driver
+
+	commonInit(s *state.State, pool *api.StoragePool, poolID int64, volume *api.StorageVolume)
+	init() error
+
+	UsesThinpool() bool
+	GetBlockFilesystem() string
+}
+
+// Driver represents a storage backend.
+type Driver interface {
+	GetVersion() string
+
+	StoragePoolCheck() error
```

Comment:
@monstermunchkin as these functions are part of an interface in the storage package, you could remove the "Storage" prefix from these function definitions to reduce "stuttering", like the Volume ones below.

Reply:
Makes sense.